shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/controler/CarminDataApi.java

```diff
@@ -0,0 +1,49 @@
+/**
+
+ * Shanoir NG - Import, manage and share neuroimaging data
+ * Copyright (C) 2009-2019 Inria - https://www.inria.fr/
+ * Contact us on https://project.inria.fr/shanoir/
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+package org.shanoir.ng.dataset.controler;
+
+import io.swagger.annotations.*;
+import org.shanoir.ng.dataset.model.carmin.GetPathResponse;
+import org.shanoir.ng.dataset.model.carmin.Path;
+import org.shanoir.ng.dataset.model.carmin.UploadData;
+import org.shanoir.ng.shared.exception.RestServiceException;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import java.io.IOException;
+
+/**
+ * @author Alae Es-saki
+ */
+@Api(value = "carmin-data")
+@RequestMapping("/carmin-data")
+public interface CarminDataApi {
+
+    @ApiOperation(value = "Get content or information for a given path", notes = "Download a file (or a directory) or retun information about a specific path. The response format and content depends on the mandatory action query parameter (see the parameter description). Basically, the \"content\" action downloads the raw file, and the other actions return various information in a JSON record.", tags={  })
+    @ApiResponses(value = {
+            @ApiResponse(code = 200, message = "successful response. If the action is \"content\", the raw file (or a tarball) is returned, with the according mime type. Otherwise a json response a returned", response = GetPathResponse.class),
+
+            @ApiResponse(code = 200, message = "A functional or internal error occured processing the request") })
```

Comment:
Hi Alae, I hope you are doing fine. Could you please adapt the return code of the error to 500? Thank you and with kind regards, Michael